tests/runtest.sh

```diff
@@ -41,6 +41,9 @@ run_test_dir()
     if [ "${__mode}" = "Cpp" ]; then
         __extra_args="${__extra_args} /p:NativeCodeGen=cpp"
     fi
+    if [ "${__mode}" = "Wasm" ]; then
```

Comment:
Wrt. which tests run and when:
The existing logic in runtest.sh was a little confusing with multiple nested if (!...) blocks. I cleaned that up (to be able to add the wasm logic without complicating it further) but might have missed something.
The logic is here:
https://github.com/dotnet/corert/pull/5297/files#diff-e99015da4fdea5ed8c0b58186a5e4b70R343

It seems to me, that when CoreRT_TestCompileMode is empty (default), it should run ryujit and cpp tests.
When it's ryujit, cpp or wasm, only run those tests.

If we want to run wasm tests when not building for wasm, wouldn't that fail since there wouldn't be a wasm native target package?

Reply:
There definitely is something I'm missing here...
I thought the wasm native package, bitcode files etc., was *only* built when running `build.sh|cmd wasm` and not when just running `build.cmd debug`.
But I can see that the Windows 'debug' build is running the wasm, so it must build two flavors of the runtime? How is that hooked up? Do we need to do the same for OSX?

Reply:
The default Windows build is just running the compiler. It is not actually running the webassembly output, nor building WebAssembly version of the runtime. You can see "Emscripten not found, not linking WebAssembly" message in the output that explains it.

This was done to get some cheap coverage in the JIT. The compiler is target independent (we use runtime checks instead of ifdefs), so telling it to produce wasm instead of native code is just a matter of passing a different switch to it.

I think it is nice-to-have for OSX/Linux. I do not think it is required. It is likely that this will stop working at some point in future as we get more complex tests compiling that will require a WebAssembly-specific build of the runtime or libraries.